rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -650,6 +668,20 @@ void Player::create_control_services()
     {
       response->success = set_rate(request->rate);
     });
+  srv_play_ = create_service<rosbag2_interfaces::srv::Play>(
+    "~/play",
+    [this](
+      rosbag2_interfaces::srv::Play::Request::ConstSharedPtr request,
+      rosbag2_interfaces::srv::Play::Response::SharedPtr response)
+    {
+      play_options_.start_offset =
+      static_cast<rcutils_time_point_value_t>(RCUTILS_S_TO_NS(request->start_offset.sec)) +
+      static_cast<rcutils_time_point_value_t>(request->start_offset.nanosec);
+      play_options_.playback_duration = rclcpp::Duration(
+        request->playback_duration.sec, request->playback_duration.nanosec);
+      play();
```

Comment:
What if we are already in playback? How this service call is going to work in t his case?

I would suggest to add `atomic_bool is_in_playback_` flag inside Player class and change API for `Player::play()` to returning **false** if we are already in playback.

Reply:
@MichaelOrlov what would it be the required interaction with the other Player APIs?

I would like to understand what would it be the expected behavior for APIs like pause(), resume(), seek() and play_next() based on this "new state". Should they care about or shall we use this flag just as a guard to avoid repetitive calls to play() that interfere one with the other?

Reply:
@agalbachicar IMO. We shall use this flag just as a guard to avoid repetitive calls to play() that interfere one with the other.
As regards to the your worries about other API:
* `play_next()` - Already handle this case
* `seek()` irrelevant to the `play()`
* `resume` irrelevant to the `play()` since it triggering flag inside internal clock abstraction class
* `pause()` irrelevant to the `play()` since it triggering flag inside internal clock abstraction class